freshmaker/pyxis.py

```diff
@@ -0,0 +1,253 @@
+
+        :param dict params: parameters to add to GET request
+        """
+        request_params = copy.deepcopy(params)
+        organization = conf.pyxis_index_image_organization
+        request_params["filter"] = "organization==" + organization
+
+        entity = "operators/indices"
+        indices = self._pagination(entity, request_params)
+        return indices
+
+    def _get_bundles_per_index_image(self, index_images, params={}):
+        """
+        Get bundle images for all index images
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: bundle images per index image
+        :rtype: dict
+        """
+        request_params = copy.deepcopy(params)
+        bundles_per_index_image = {}
+        # we need 'bundle_path_digest' to find ContainerImage of that bundle
+        include_fields = ['data.channel_name', 'data.version',
+                          'data.related_images', 'data.bundle_path_digest']
+        request_params['include'] = ','.join(include_fields)
+        for index_image in index_images:
+            path = index_image.get('path', '')
+            if not path:
+                continue
+            request_params['filter'] = f"source_index_container_path=={path}"
+            bundles = self._pagination('operators/bundles', request_params)
+            bundles_per_index_image[path] = bundles
+
+        return bundles_per_index_image
+
+    def get_bundles_per_channel_per_index_image(self, index_images, params={}):
+        """
+        Get bundle images per channel per index image
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: bundle images per channel per index image
+        :rtype: dict
+        """
+        bundles_per_index_image = \
+            self._get_bundles_per_index_image(index_images, params)
+
+        ret = {}
+        for index_image, bundles in bundles_per_index_image.items():
+            bundle_per_channel = {}
+            # get latest versions of bundle images per channel
+            for bundle in bundles:
+                channel = bundle.get('channel_name')
+                if channel in bundle_per_channel:
+                    new_ver = bundle.get('version')
+                    old_ver = bundle_per_channel[channel].get('version')
+                    # Check if the right format of version is used
+                    try:
+                        if version.parse(new_ver) > version.parse(old_ver):
+                            bundle_per_channel[channel] = bundle
+                    except version.InvalidVersion as e:
+                        log.warning("Other format than SemVer is used")
+                        log.warning(repr(e))
+                else:
+                    # Check if the right format of version is used
+                    try:
```

Comment:
Since you always have to parse `new_ver` and that's the only one you need to validate for loop iteration, could you move the try/except outside of the if statement so you don't need to duplicate the code?